webpack/assets/javascripts/react_app/redux/actions/notifications/index.js

```diff
@@ -0,0 +1,54 @@
+import {
+  NOTIFICATIONS_GET_NOTIFICATIONS,
+  NOTIFICATIONS_TOGGLE_DRAWER,
+  NOTIFICATIONS_SET_EXPANDED_GROUP,
+  NOTIFICATIONS_MARK_AS_READ
+} from '../../consts';
+import { notificationsDrawer } from '../../../common/sessionStorage';
+import API from '../../../API';
+
+export const getNotifications = url => dispatch => {
+  API.get(url).then(response => dispatch({
```

Comment:
in the current implementation there are two addition cases which are missing:
1. when there is a failure, stop pulling notifications (e.g. session timeout, or server 500 error).
2. when you are not in focus, we dont want to pull data from the server when there is no one to consume it.